### PR TITLE
chore: Update typescript & q versions

### PR DESCRIPTION
**Pull request by smoralesd**
#### Description of changes

Update typescript & q version.
Update types:
- Using `MessageEventSource` instead of `Window`
- Using `HTMLCollectionOf` instead of `NodeListOf`
- Casting `'*'` as `string & Transferable[]` (on `postMessage`)

All the type changes came from typescript type updates.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`npm run test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`npm run precheckin`)
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

**Comment by Shobhit1** [comment]
I think we should wait till #647 is merged as we got rid of `package-lock` 😄

**Comment by smoralesd** [comment]
> I think we should wait till #647 is merged as we got rid of `package-lock` 😄

Doesn't seem necessary to explicitly wait. Whoever goes second needs to update his PR.

File: src/injected/tab-stops-listener.ts [hunk 1]
```diff
@@ -131,7 +131,7 @@ export class TabStopsListener {
     @autobind
     private onStopListenToTabStops(): void {
         this.removeListeners();
-        const iframes: NodeListOf<HTMLIFrameElement> = this.getAllFrames();
```
Comment (lisli1) [inline src/injected/tab-stops-listener.ts]: I noticed that in other places, `NodeListof<HTMLIFrameElements>` is replaced with `HTMLCollectionOf<HTMLIFrameElements>`, so I am wondering why that is not the case here?

File: src/tests/unit/tests/injected/frameCommunicators/window-message-handler.test.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ import { FrameMessageResponseCallback, WindowMessageHandler } from '../../../../
 import { WindowMessageMarshaller } from '../../../../../injected/frameCommunicators/window-message-marshaller';
 
 describe('WindowMessageHandlerTests', () => {
+    const starOrigin = '*' as (string & Transferable[]);
```
Comment (waabid) [inline src/tests/unit/tests/injected/frameCommunicators/window-message-handler.test.ts]: What is this Transferable[] about? Should we establish a type for this; seems to be used in a few places.

Comment (smoralesd) [reply src/tests/unit/tests/injected/frameCommunicators/window-message-handler.test.ts]: Not sure what it is, it came up from an error while updating typescript so I'm just casting to the type to fix all errors.

File: src/injected/tab-stops-listener.ts (hunk 1 above)
Comment (smoralesd) [reply src/injected/tab-stops-listener.ts]: This seems outdated, I don't see the explicit type for the const

File: src/tests/unit/tests/injected/frameCommunicators/window-message-handler.test.ts (hunk 2 above)
Comment (smoralesd) [reply src/tests/unit/tests/injected/frameCommunicators/window-message-handler.test.ts]: `Transferable` documentation [here](https://developer.mozilla.org/en-US/docs/Web/API/Transferable)

**Comment by smoralesd** [comment]
@microsoft/accessibility-insights-web-code-owners 
@dbjorge was looking at this issue and found out we're getting "out of memory" issues. This are related to typescript 3.3.x and 3.4.x so I'm closing this PR for now as it requires some more investigation.

**Pull request closed without merging**